src/map/script.cpp

```diff
@@ -26957,6 +26957,37 @@ BUILDIN_FUNC(has_autoloot) {
 	return SCRIPT_CMD_SUCCESS;
 }
 
+// ===================================
+// *autoloot(<rate>{, <char_id>});
+// This command sets the rate of autoloot (10000 = 100%).
+// Returns true on success and false on failure.
+// Note: It will toggle existing autoloot settings if rate value aren't provided.
+// ===================================
+BUILDIN_FUNC(autoloot) {
+	map_session_data *sd;
+
+	if (!script_charid2sd(3, sd)) {
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int rate = 0;
+	if (script_hasdata(st, 2))
+		rate = script_getnum(st, 2);
+	else if (sd->state.autoloot <= 0)
+		rate = 10000;
```

Comment:
```suggestion
```

Reply:
It was meant to turn off the autoloot if user already autoloot turned on.

Reply:
I think what you really mean is the exact opposite. This means if it was turned off, then it should be turned on.

Reply:
yeah
if existing setting is off, then it should turn on (100%)
if existing setting is on, then it should turn off (0%)

```
autoloot(); // toggle on/off depend on existing autoloot
autoloot(0); // 0% / off
autoloot(100); // 1%
autoloot(5000); // 50%
autoloot(10000); // 100%
```

Reply:
How about
```suggestion
	else
		rate = (sd->state.autoloot > 0 ? 0 : 10000);      // toggle on/off depend on existing autoloot
```

Reply:
@Atemo i dont mind this, but it should match the documentation.
Either the default value is 100% or the default value is 100% (if currently off) or 0% (if currently on).

Reply:
updated